File: C/pr_2.c
#include <stdio.h>

void space(int n, int i) {
    for (int j = 1; j <= n - i; j++) {
        printf(" ");
    }
}

void main() {
    int n = 4;
    for (int i = 1; i <= n; i++) {
        space(n, i);
        for (int j = 1; j <= i; j++) {
            printf("%c ", i + 64);
        }
        printf("\n");
    }
}

File: Python/pr_2.py
#!/bin/python3

"""
   A
  B B
 C C C
D D D D
"""

def space(n, i):
    for j in range(1, n + 1 - i):
        print(" ", end="")

def main():
    n = 4
    for i in range(1, n + 1):
        space(n, i)
        for j in range(1, i + 1):
            print(chr(i + 64), end=" ")
        print()

if __name__ == "__main__":
    main()
